codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt

```diff
@@ -376,6 +376,13 @@ private fun Element.changeInto(tagName: String) {
     replaceWith(Element(tagName).also { elem -> elem.appendChildren(childNodesCopy()) })
 }
 
+/** Write an `impl` block for the given symbol */
+fun RustWriter.implBlock(symbol: Symbol, block: Writable) {
```

Comment:
Yeah there's tons of places in the server where we could use this syntax sugar, thanks for making use of it whenever possible.